libs/designsystem/src/lib/components/modal/services/modal.controller.ts

```diff
@@ -17,14 +18,34 @@ export class ModalController {
   constructor(
     private modalHelper: ModalHelper,
     private actionSheetHelper: ActionSheetHelper,
-    private alertHelper: AlertHelper
-  ) {}
+    private alertHelper: AlertHelper,
+    private modalOutlet: ModalOutlet
+  ) {
+    this.resolveModal();
+  }
+
+  private resolveModal() {
+    this.modalOutlet.resolve$.subscribe(() => {
+      if (this.overlays.length === 0) this.showModal({ title: '' });
+    });
+  }
 
   public async showModal(config: ModalConfig, onClose?: (data?: any) => void): Promise<void> {
     if (config.hasOwnProperty('dim')) {
       console.warn('ModalConfig.dim is deprecated - please remove from your configuration.');
     }
-    await this.showAndRegisterOverlay(() => this.modalHelper.showModalWindow(config), onClose);
+
+    const onCloseOveride = (data?: any) => {
```

Comment:
Consider naming:
```suggestion
    const onCloseWrapper = (data?: any) => {
```
OR
```suggestion
    const onCloseInternal = (data?: any) => {
```